File: packages/capacitor-plugin/android/src/main/java/io/ionic/backgroundrunner/plugin/RunnerConfig.kt
package io.ionic.backgroundrunner.plugin;

import org.json.JSONObject
import java.lang.Exception

data class RunnerConfig( val label: String,
                         val src: String,
                         val autoSchedule: Boolean,
                         val repeats: Boolean,
                         var event: String,
                         val interval: Int?) {


    companion object {
        fun fromJSON(configObject: JSONObject): RunnerConfig {
            val labelStr = configObject.getString("label")
            if (labelStr.isNullOrEmpty()) {
                throw Exception("runner label is missing or invalid")
            }

            val srcStr = configObject.getString("src")
            if (srcStr.isNullOrEmpty()) {
                throw Exception("runner source file path is missing or invalid")
            }

            return RunnerConfig(
                labelStr,
                srcStr,
                configObject.optBoolean("autoStart", false),
                configObject.optBoolean("repeat"),
                configObject.optString("event"),
                configObject.optInt("interval", 0)
            )
        }
    }

}

File: packages/capacitor-plugin/ios/Sources/CapacitorBackgroundRunner/RunnerConfig.swift
import Foundation
import Capacitor

public struct RunnerConfig {
    let label: String
    let src: String
    var autoSchedule: Bool
    var event: String
    let repeats: Bool
    let enableWatchConnectivity: Bool

    let interval: Int?

    public init(from jsObject: JSObject) throws {
        guard let label = jsObject["label"] as? String else {
            throw BackgroundRunnerPluginError.invalidRunnerConfig(reason: "runner label is missing or invalid")
        }

        guard let src = jsObject["src"] as? String else {
            throw BackgroundRunnerPluginError.invalidRunnerConfig(reason: "runner source file path is missing or invalid")
        }

        let event = jsObject["event"] as? String ?? ""
        let repeats = jsObject["repeat"] as? Bool
        let interval = jsObject["interval"] as? Int
        let autoStart = jsObject["autoStart"] as? Bool

        self.label = label
        self.src = src
        self.event = event
        self.repeats = repeats ?? false
        self.interval = interval
        self.autoSchedule = autoStart ?? false
        self.enableWatchConnectivity = jsObject["enableWatchConnectivity"] as? Bool ?? false
    }
}
